fv3gfs-util/fv3gfs/util/partitioner.py

```diff
@@ -568,22 +592,21 @@ def global_extent(self, rank_metadata: QuantityMetadata) -> Tuple[int, ...]:
             rank_metadata.dims, rank_metadata.extent, self.layout
         )
 
-    def subtile_extent(self, cube_metadata: QuantityMetadata) -> Tuple[int, ...]:
+    def subtile_extent(self, cube_metadata: QuantityMetadata, rank: int, overlap: bool) -> Tuple[int, ...]:
```

Comment:
I think overlap should always be `True` here. Subtile "extent" inherently should include overlapping points, it's when you're slicing a full tile and want to ensure subtile data doesn't overlap that you need the subtile "slice" not to include overlap.

Reply:
Changed all overlaps to be `True` by default.

Reply:
Why is overlap an argument? When should it be `False`?

Reply:
Overlap as an argument mirrors `subtile_slice`, where it is also one. The way the code currently calculates the extents of a subtile incorporates `overlap` as an argument. This function uses the same mechanism, and it assumes that `overlap` _can_ be `False`.

Reply:
But when `subtile_slice` calls `subtile_extent`, it will always pass `overlap=True`. It will give an incorrect answer if it passes `overlap=False`. This is not a value that should be modified inside this function (unless you can describe a use case where it should be modified to False).

Reply:
Another way to put it is this - if you are going to keep this as an argument, what would you put in the docstring to describe what it does, and when you would want to set it to what value?

Reply:
`subtile_slice` never calls `subtile_extent`, it just has an int tuple named like that. Which should probably be changed to `subtile_extents` instead to avoid confusion.

Reply:
Sorry, I meant to say "when `subtile_extent` calls `subtile_slice`. subtile_extent returns an int tuple, and "extent" is the correct name for an int tuple describing the domain size.

Reply:
Fair enough. `overlap` isn't needed, you're right.